### DISCUSSION TITLE
Docker "WARN: Failed to obtain location/ISP info. Got: Error: stun inaccessible" error

### DISCUSSION THREAD

**Discussion by rgraham7503** [hyperjumptech/monika#847] [General]
Having an issue with the latest docker version. Getting this error as show in image - We had this issue previously and I thought it was fixed? "WARN: Failed to obtain location/ISP info. Got: Error: stun inaccessible" 
This is on macOS and pulled the latest docker image
![image](https://user-images.githubusercontent.com/46932722/183250178-e3747543-eb66-47af-99d7-84eeb05b5ee3.png)
This is the docker run command as shown in documentation and I verified command ran is the same as one listed in a git reply.
`docker run --name monika_interactive -v monitor.yml:/config/monika.yml -it hyperjump/monika:latest monika -c /config/monika.yml --prometheus 3001
`
`
`
showing the latest pull:

`latest: Pulling from hyperjump/monika
df9b9388f04a: Pull complete 
622e2b598d8a: Pull complete 
f7c8a32a53f2: Pull complete 
7da04ed7d1ef: Pull complete 
9cf5699a0caf: Pull complete 
a8fb0c6fbff5: Pull complete 
a5716a802d5f: Pull complete 
Digest: sha256:29d27ce37e44aa2f6dd6546640c9b932eac20dd2bb1d83a156965bf88f85c2e0
Status: Downloaded newer image for hyperjump/monika:latest`

**Comment by rgraham7503** [comment]
I did try adding --net=host as requested in a previous post but it did not understand the command..
`CLIError: Configuration file not found: --net=host`
`docker run --name monika_interactive -v monitor.yml:/config/monika.yml -it hyperjump/monika:latest monika -c /config/monika.yml --net=host --prometheus 3001`

**Reply by nicnocquee** [reply]
I think the `--net=host` should be before `monika -c` because it's not monika's flag but the docker's flag.

I tried the latest docker in my mac Monterey 12.4 (21F79) 

<img width="1368" alt="Screenshot 2022-08-06 at 17 37 00" src="https://user-images.githubusercontent.com/311343/183255957-98d3a7b0-8973-4673-9a36-6ae346aaf1e3.png">

Also in Ubuntu 20.04.2 LTS

<img width="1223" alt="Screenshot 2022-08-06 at 17 35 05" src="https://user-images.githubusercontent.com/311343/183255979-fe625e0b-28ce-4630-b4d7-aab32e6b0120.png">

**Reply by rgraham7503** [reply]
I tried and still got the error
`docker run --name monika --net=host -v monitor.yml:/config/monika.yml -it hyperjump/monika:latest monika -c /config/monika.yml --prometheus 3001`
![image](https://user-images.githubusercontent.com/46932722/183259650-2405ed0d-9379-44d2-adc6-6608ffc6bd0a.png)

**Reply by sapiderman** [reply]
@rgraham7503 

Can you please let the logs run a little longer? Want to check if any of your (2) probes starts and queries properly.
The STUN warnings should not be fatal.

**Reply by rgraham7503** [reply]
@sapiderman I let it run 3 minutes...same behavior..
![image](https://user-images.githubusercontent.com/46932722/183406669-12a7343e-39c4-4059-9095-d944624eb3df.png)

**Reply by sapiderman** [reply]
Great! Thanks for the logs @rgraham7503 ... Looking into it further.

**Comment by sapiderman** [comment]
Hmm getting the same result on the console/cli version

This is with -s 0 (poll STUN just once):

![monika-blocked-pauses](https://user-images.githubusercontent.com/964106/183807711-455e918d-b525-494a-a75e-69f5d463aaef.png)

This is with continuous polling:

![monika-blocked-timed-out](https://user-images.githubusercontent.com/964106/183807806-1c82b93d-03b2-403d-94cd-177b8de90d17.png)

**Comment by sapiderman** [comment]
@rgraham7503 

Checkout the latest release : https://github.com/hyperjumptech/monika/releases/tag/v1.12.0
Fixed the STUN blocking issue. Now you can disable it with a `-s -1` flag. Check the docs and let us know if you've got questions!

Regards

**Comment by rgraham7503** [comment]
Confirmed! This did fix the issue, however now I am not able to get the events into Prometheus. This could be some weird macos/docker issue I must be having. Could you validate --prometheus 3001 still works? It shows you can scrape metrics but when I check prometheus it shows it cant connect to localhost:3001
`INFO: Updating config
INFO: You can scrape Prometheus metrics from http://localhost:3001/metrics`
![image](https://user-images.githubusercontent.com/46932722/187914305-9224f347-9f99-4c9e-950f-dada3b7c9d96.png)

**Reply by rgraham7503** [reply]
Ignore, I tried to run monika -c and feed to prometheus and same issue. Something wrong with my Prometheus/docker/mac. Very frustrating. Thanks for the fix though! working great, now to get prometheus to work right
